.github/workflows/publish.yml

```diff
@@ -7,56 +7,67 @@ on:
   release:
     types: [published]
 
+env:
+  VERSION: ${{ github.event.release.tag_name }}
+
 jobs:
   publish:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+
       - uses: actions/setup-node@v4
         with:
           node-version: "22"
+
       - name: Generate source archive
         run: npm run archive
-        env:
-          VERSION: ${{ github.event.release.tag_name }}
+
+      - name: Upload source archive
+        uses: actions/upload-artifact@v4
+        with:
+          name: codecov-browser-extension-${{ github.event.release.tag_name }}.tar.gz
+          path: codecov-browser-extension-${{ github.event.release.tag_name }}.tar.gz
+
       - name: Install dependencies
-        run: npm ci
-      - name: Build extension
+        run: npm install
+
+      - name: Build Chrome extension
         run: npm run build
         env:
           SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
-          VERSION: ${{ github.event.release.tag_name }}
-      - name: Prepare Firefox extension
```

Comment:
key thing is we need to use the build:firefox script instead of this manual move. Reason being the builds are actually different now with the IS_FIREFOX env var